ingestion-beam/src/main/java/com/mozilla/telemetry/amplitude/ParseAmplitudeEvents.java

```diff
@@ -192,30 +192,32 @@ List<String[]> readAllowedEventsFromFile() throws IOException {
       throw new IllegalArgumentException("File location must be defined");
     }
 
-    try (InputStream inputStream = BeamFileInputStream.open(eventsAllowListPath);
-        InputStreamReader inputStreamReader = new InputStreamReader(inputStream);
-        BufferedReader reader = new BufferedReader(inputStreamReader)) {
-      singletonAllowedEvents = new ArrayList<String[]>();
+    synchronized (ParseAmplitudeEvents.class) {
```

Comment:
When multiple threads arrive here at the same time at startup, they'll proceed to enter the synchronized block in turn. As a result they'll read the file multiple times.
I think it's safe from concurrency point of view as other threads will see the list fully initialized (because of the synchronized block), but if we wanted to make sure that this file is read only once we'd need to wrap this in another null check:
```suggestion
    synchronized (ParseAmplitudeEvents.class) {
       if (singletonAllowedEvents == null) {
         ...
```
and make singletonAllowedEvents `volatile`.